### DISCUSSION TITLE
Magic link with local dev environment

### DISCUSSION THREAD

**Discussion by ProjectMoon** [supabase/supabase#1044] [Questions] [answered]
Is there a way to find the magic link login URL when using the local dev environment set up with the supabase CLI? The API call completes fine, but obviously no email is sent. Is there somewhere in Docker logs I could find it? I don't see a magic link URL in supabase-auth, for example.

**Comment by kiwicopple** [comment]
Hmm, we don't actually add valid SMTP credentials for local dev, so the emails won't get sent. 

In `gotrue-js` we use `inbucket` for testing: https://github.com/supabase/gotrue-js/blob/5690b208ee6f0fa5cf0cf55f5251a10a2bfe5e03/infra/docker-compose.yml#L74

Perhaps you can open an issue in our CLI repo so that we can add local mail support ? https://github.com/supabase/cli

**Comment by ProjectMoon** [comment]
Yes I will do that. With the firebase cli, it just prints the login link to the terminal in the logs of the emulator. So something similar in the auth container logs would probably suffice.

**Comment by loong** [comment, marked as answer]
Inbucket is an email testing service and runs as part of the local supabase setup if you use the CLI.

```
$ supabase start
Started supabase local development setup.          
                                                                    
         API URL: http://localhost:54321
     GraphQL URL: http://localhost:54321/graphql/v1
          DB URL: postgresql://postgres:postgres@localhost:54322/postgres
      Studio URL: http://localhost:54323
    Inbucket URL: http://localhost:54324       <------
      JWT secret: [secret key]
        anon key: [anon key]
service_role key: [secret key]
```

You can open Inbucket at http://localhost:54324. And looks like this below. There you can open the email and click on the magic link.
![image](https://github.com/supabase/supabase/assets/1732217/76110a84-6034-4b81-a696-d4254b6e477f)


Side note: for magic links to be sent locally, you also have to make sure `enable_confirmations` is set to true in the `supabase/config.toml` file.

**Reply by taufik-nurhidayat** [reply]
Thank you for this answer, it's help me
